src/cpp/session/modules/clang/RCompilationDatabase.cpp

```diff
@@ -1178,10 +1178,17 @@ std::vector<std::string> RCompilationDatabase::baseCompilationArgs(bool isCpp) c
    args.push_back("-nostdinc");
 
    // add Rtools arguments
-   auto rtInfo = rToolsInfo();
-   auto rtArgs = rtInfo.clangArgs(isCpp);
-   args.insert(args.end(), rtArgs.begin(), rtArgs.end());
+   //auto rtInfo = rToolsInfo();
+   //auto rtArgs = rtInfo.clangArgs(isCpp);
+   //args.insert(args.end(), rtArgs.begin(), rtArgs.end());
```

Comment:
We can skip calling this function entirely in favor of the compiler reported paths from `where gcc.exe ...`

Reply:
If possible, I'd prefer to keep the old code alive for older Rtools installations, just because we "know" it works and there's a possibility that this doesn't do what we expect with older Rtools installations.

In practice, that means using this code for R < 4.2.0, and the new "discover system includes" path for R >= 4.2.0 (with a comment to justify). Do you think that is reasonable?